src/java/org/transmartproject/db/dataquery/highdim/snp_lz/GzipFieldTokenizer.java

```diff
@@ -144,4 +152,78 @@ public T apply(Reader reader) {
 
         });
     }
+
+    private void appendString(List<String> l, String s) {
+        if (l.size() > expectedSize - 1) {
+            throw new InputMismatchException(
+                    "Got more tokens than the " +
+                            expectedSize + " expected");
+        }
+        l.add(s);
+    }
+
+    /**
+     * Read characters from the space-delimited stream one-by-one and 
+     * use a StringBuilder to build the read values (String values of any size),
+     * appending them to a list of Strings.
+     *
+     * @throws InputMismatchException iff the number of values read &ne; <var>expectedSize</var>.
+     * @return a list of strings.
+     */
+    public List<String> asStringList() throws IOException, InputMismatchException, SQLException {
+        final char[] buffer = new char[1];
+        return withReader(new Function<Reader, List<String>>() {
+            public List<String> apply(Reader r) {
+                List<String> res = new ArrayList<String>(expectedSize);
+                StringBuilder builder = new StringBuilder();
+                int n;
+                char c;
+                try {
+                    while ((n = r.read(buffer)) > 0) {
```

Comment:
Since you're reading byte by byte, you might as well use the single-byte read().